octoprint_octolapse/render.py

```diff
@@ -568,6 +591,39 @@ def _render(self):
 
         self._on_complete()
 
+    def _preprocess_images(self, processed_filepath):
+        first_timestamp = None
+        for i in itertools.count():
+            input_path = "{0}{1}".format(self._capture_dir, self._capture_file_template) % i
+            if not os.path.isfile(input_path):
+                break
+            # Get file creation time, or failing that, last file modification time.
+            timestamp = os.path.getctime(input_path) or os.path.getmtime(input_path)
+            if first_timestamp is None:
+                first_timestamp = timestamp
+            current_time = datetime.fromtimestamp(timestamp).strftime("%Y-%m-%d %H:%M:%S")
+            time_elapsed = str(timedelta(seconds=round(timestamp - first_timestamp)))
+
+            image = Image.open(input_path)
+            # Flip image if configured.
+            if self._rendering.flip_h:
```

Comment:
Rotation is currently also done in snapshot.py (around line 442), but only for webcam images (camera transposition options).  I think this was just an oversight on my part when I added the camera transpose option.  Maybe that is confusing?  I see the benefit of waiting to rotate until the end of the print, but I think it would be useful to make a note of this in the rendering/camera settings page somehow.  Thoughts or ideas?

Reply:
I just replaced the rotation in the `ffmpeg` pipeline with this rotation. I'm not sure what you mean by rotation being done in `snapshot.py`; I'll have to check that out.

I think rotating at snapshot-time is fine though. I'll have to improve my test images to test that (I use random blocks of colour :P